linker.go

```diff
@@ -169,6 +169,10 @@ func fixupAndValidate(insns asm.Instructions) error {
 			return fmt.Errorf("instruction %d: map %s: %w", iter.Index, ins.Reference(), asm.ErrUnsatisfiedMapReference)
 		}
 
+		if ins.IsFunctionReference() && haveBPFToBPFCalls() != nil {
```

Comment:
I'm a bit torn about this to be honest, refusing to load is a pretty big hammer. Feature tests are kind of brittle, so the less we rely on them the better.

It'd probably also be better to only call haveBPFToBPFCalls once per program instead of once per bpf2bpf call? It's guarded by a RWMutex, but its a mutex after all.

Reply:
> It'd probably also be better to only call haveBPFToBPFCalls once per program instead of once per bpf2bpf call? It's guarded by a RWMutex, but its a mutex after all.

True, but I didn't want to call it if the prog didn't contain any func references.

> refusing to load is a pretty big hammer

Agreed, maybe it's better to use the probe as a hint when a prog load fails, like we do for EPERM. I'll mull this over for a bit.